packages/eas-cli/src/commands/channel/list.ts

```diff
@@ -108,13 +173,41 @@ export default class ChannelList extends Command {
     });
 
     for (const channel of channels) {
-      // TODO (cedric): refactor when multiple branches per channel are available
-      const branch = channel.updateBranches[0];
-      const update = branch.updates[0];
+      const { branchMapping, isRollout, rolloutPercent } = getBranchMapping(channel.branchMapping);
+      const rolloutMapping: string[] = [];
+      let update:
+        | null
+        | (Omit<Partial<Update>, 'actor'> & {
+            actor?: null | Pick<Actor, 'id' | 'firstName'>;
+          }) = null;
+
+      for (const index in branchMapping.data) {
+        if (parseInt(index, 10) > 1) {
+          throw new Error('Branch Mapping data must have length less than or equal to 2.');
+        }
+
+        const { branchId } = branchMapping.data[index];
+        const branch = channel.updateBranches.find(branch => branch.id === branchId);
+        if (!branch) {
+          throw new Error('Branch mapping is pointing at a missing branch.');
+        }
+
+        if (!isRollout) {
+          rolloutMapping.push(branch.name);
+        } else {
+          rolloutMapping.push(
+            parseInt(index, 10) === 0
+              ? `${branch.name} (${rolloutPercent! * 100}%)`
+              : `${branch.name} (${(1 - rolloutPercent!) * 100}%)`
+          );
+        }
+
+        update = branch.updates[0];
```

Comment:
@jkhales is this correct?

Reply:
Seems like it should be fine, if no update we just get undefined/blank in the output.